app/src/main/java/fr/free/nrw/commons/upload/FileMetadataUtils.java

```diff
@@ -0,0 +1,102 @@
+package fr.free.nrw.commons.upload;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+import io.reactivex.Observable;
+import timber.log.Timber;
+
+import static androidx.exifinterface.media.ExifInterface.*;
+
+public class FileMetadataUtils {
+
+    public static Observable<String> getTagsFromPref(String pref) {
+        Timber.d("Retuning tags for pref:%s", pref);
+        switch (pref) {
+            case "Author":
+                return Observable.fromArray(TAG_ARTIST, TAG_CAMARA_OWNER_NAME);
+            case "Copyright":
+                return Observable.fromArray(TAG_COPYRIGHT);
+            case "Location":
+                return Observable.fromArray(TAG_GPS_LATITUDE, TAG_GPS_LATITUDE_REF,
+                        TAG_GPS_LONGITUDE, TAG_GPS_LONGITUDE_REF,
+                        TAG_GPS_ALTITUDE, TAG_GPS_ALTITUDE_REF);
+            case "Camera Model":
+                return Observable.fromArray(TAG_MAKE, TAG_MODEL);
+            case "Lens Model":
+                return Observable.fromArray(TAG_LENS_MAKE, TAG_LENS_MODEL, TAG_LENS_SPECIFICATION);
+            case "Serial Numbers":
+                return Observable.fromArray(TAG_BODY_SERIAL_NUMBER, TAG_LENS_SERIAL_NUMBER);
+            case "Software":
+                return Observable.fromArray(TAG_SOFTWARE);
+            default:
+                return null;
+        }
+    }
+
+    /**
+     * Removes all XMP data from the input file and writes the rest of the image to a new file.
+     *
+     * This works by black magic. Please read the JPEG section of the XMP Specification Part 3 before making changes.
+     * https://wwwimages2.adobe.com/content/dam/acom/en/devnet/xmp/pdfs/XMP%20SDK%20Release%20cc-2016-08/XMPSpecificationPart3.pdf
+     *
+     * @param inputPath the path of the input file
+     * @param outputPath the path of the new file
+     */
+    public static void removeXmpAndWriteToFile(String inputPath, String outputPath) {
```

Comment:
There are a lot of [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)) in this method. It would be great if you could extract these numbers to named constants. 

Also, the method itself is quite large. Would it be possible to break down the method to smaller logical parts?

Reply:
It turned out that XMP doesn't work here so it was removed